app/shared/design-system/components/primitives/auto-input.tsx

```diff
@@ -0,0 +1,205 @@
+export class AutoInput extends EventTarget {
+	// PRIVATE
+	#selectTarget = null;
+	#activeTarget = null;
+	#activeEvents = ['keydown', 'focusin', 'focusout', 'pointerover', 'pointerout', 'pointerup', 'pointerdown'];
+	#passiveEvents = ['pointermove'];
+	#resizeCursor = 'ns-resize';
+	#pointerCursor = 'pointer';
+	#pointerCapture = false;
+	#ownerDocument = globalThis.document;
+	#canvasContext = globalThis.document?.createElement('canvas').getContext('2d');
+	#typeOfValue(value) {
+		if (/^([-+]?\d*\.?\d+)$/.test(value)) return 'number';
+		if (/^(%|cm|mm|in|px|pt|pc|em|ex|ch|rem|vw|vh|vmin|vmax|vb|vi|svw|svh|lvw|lvh|dvw|dvh|lh|rlh)$/.test(value)) return 'unit';
+		if (/^(\w+)$/.test(value)) return 'keyword';
+		if (/^(\w+\()/.test(value)) return 'function';
+		return '';
+	}
+	#measureEvent(value, event, target) {
+		const {fontSize, fontFamily, fontWeight, textAlign, paddingLeft, paddingRight} = getComputedStyle(target);
+		this.#canvasContext.font = `${fontWeight} ${fontSize} ${fontFamily}`;
+		const eventOffset = event.offsetX;
+		let charsOffset = 0;
+		let pixelPadding = 0
+		switch (textAlign) {
```

Comment:
whats this handling? can't guess from reading

Reply:
uhh, a lot of other things can impact the measuring, e.g. letter spacing

Reply:
I mean if user defines custom letter spacing as style, same as font-family, it will impact the size of text, and I haven't seen you considering it, also letter spacing might be not the only one missing